frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -371,14 +387,26 @@ pub mod pallet {
         /// This must be called by the developer who registered the contract.
         ///
         /// Warning: After this action contract can not be assigned again.
-        #[pallet::weight(T::WeightInfo::unregister(T::MaxNumberOfStakersPerContract::get()))]
+        #[pallet::weight(T::WeightInfo::unregister())]
         pub fn unregister(
             origin: OriginFor<T>,
             contract_id: T::SmartContract,
         ) -> DispatchResultWithPostInfo {
             ensure!(!Self::pallet_disabled(), Error::<T>::Disabled);
             let developer = ensure_signed(origin)?;
 
+            let mut dapp_info =
+                RegisteredDapps::<T>::get(&contract_id).ok_or(Error::<T>::NotOperatedContract)?;
+            ensure!(
+                dapp_info.state == DAppState::Registered,
+                Error::<T>::NotOperatedContract
+            );
+            ensure!(
+                dapp_info.developer == developer,
+                Error::<T>::NotOwnedContract,
+            );
+
+            // TODO: Is this really needed? Why shouldn't same account be eligible to register multiple contracts?
```

Comment:
perhaps span protection?

Reply:
We have two levels of spam protection:
- dev needs to be whitelisted
- each registration requires reservation amount

It'd be a very expensive spam.